vendor/init.bat

```diff
@@ -220,70 +292,71 @@ exit /b
     if %verbose-output% gtr 0 echo %*
     exit /b
 
+:show_error
+    echo ERROR: %*
+    echo CMDER Shell Initialization has Failed!
+    exit /b
+
+:run_profile_d
+  if not exist "%~1" (
+    mkdir "%~1"
+  )
+  
+  pushd "%~1"
+  for /f "usebackq" %%x in ( `dir /b *.bat *.cmd 2^>nul` ) do (
+    call :verbose-output Calling "%~1\%%x"...
+    call "%~1\%%x"
+  )
+  popd
+  exit /b
+
 ::
 :: specific to git version comparing
 ::
 :read_version
-
     :: clear the variables
     set GIT_VERSION_%~1=
 
     :: set the executable path
     set "git_executable=%~2\git.exe"
+    call :verbose-output git_executable=%git_executable%
 
     :: check if the executable actually exists
     if not exist "%git_executable%" (
-        :: return a negative error code if the executable doesn't exist
+        call :verbose-output "%git_executable%" does not exist!
         exit /b -255
     )
 
     :: get the git version in the provided directory
-    for /F "delims=" %%F in ('%git_executable% --version 2^>nul') do @(
-        set "GIT_VERSION_%~1=%%F"
+    for /F "tokens=1,2,3 usebackq" %%F in (`"%git_executable%" --version 2^>nul`) do @(
+        set "GIT_VERSION_%~1=%%H"
+        call :verbose-output GIT_VERSION_%~1=%%H
     )
 
     :: parse the returned string
+    call :verbose-output Calling :validate_version "%~1" !GIT_VERSION_%~1!
     call :validate_version "%~1" !GIT_VERSION_%~1!
-
-goto :eof
+    exit /b
 
 :parse_version
-
     :: process a `git version x.x.x.xxxx.x` formatted string
     for /F "tokens=1-3* delims=.,-" %%A in ("%2") do (
         set "%~1_MAJOR=%%A"
         set "%~1_MINOR=%%B"
         set "%~1_PATCH=%%C"
         set "%~1_BUILD=%%D"
     )
-
-goto :eof
+    exit /b
 
 :validate_version
+    :: now parse the version information into the corresponding variables
+    call :parse_version %~1 %~2
 
-    :: check if we have a valid version string
-    if /I "%~2 %~3"=="GIT VERSION" (
-
-        :: now parse the version information into the corresponding variables
-        call :parse_version %~1 %~4
-
-        :: ... and maybe display it, for debugging purposes.
-        call :verbose-output Found Git Version for %~1: !%~1_MAJOR!.!%~1_MINOR!.!%~1_PATCH!.!%~1_BUILD!
-
-    ) else (
-        :: invalid format returned, use the vendored git instead
-        echo Invalid git version at "%git_executable%" detected!
```

Comment:
With the removal of this section, invalid formatted return string is not caught.

**Valid case:** if user has _any other_ executable in path accidentally named `git.exe` that won't return `git version xxxxx` on issuing `git.exe --version` on them.

IMO, `init.bat` should also verify if the return string has `git version` in it.

Reply:
fixed but slightly diff but it does what you want

Reply:
What did you change for this case?